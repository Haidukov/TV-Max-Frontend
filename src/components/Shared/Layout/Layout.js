import React from 'react';
import PropTypes from 'prop-types';

import s from './Layout.scss';
import logoUrl from '../../../assets/images/tvm-header-logo.png';

const Layout = ({ children }) => (
  <div>
    <header className={s.header}>
      <img src={logoUrl} className={s.logo} alt='TV Maze Logo' />
    </header>
    <div className={s.content}>
      {children}
    </div>
  </div>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired
};

export default Layout;
