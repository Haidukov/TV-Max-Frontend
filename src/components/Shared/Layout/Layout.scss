@import '../../../assets/styles/variables.scss';

.header {
    padding: 15px;
    background-color: $header-bg;
}

.logo {
    max-height: 150px;
    max-width: 150px;
}

.content {
    padding: 10px;
    max-width: 700px;
    margin: 0 auto;
}