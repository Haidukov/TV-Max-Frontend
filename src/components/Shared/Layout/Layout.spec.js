import React from 'react';
import { render } from '@testing-library/react';
import Layout from './Layout';

describe('<Layout />', () => {
  it('should match snapshot', () => {
    const TestComponent = () => <div>test</div>;

    const { asFragment } = render(
      <Layout>
        <TestComponent />
      </Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
