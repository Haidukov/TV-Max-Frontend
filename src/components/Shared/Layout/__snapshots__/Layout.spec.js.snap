// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="header"
    >
      <img
        alt="TV Maze Logo"
        class="logo"
        src="test url"
      />
    </header>
    <div
      class="content"
    >
      <div>
        test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
